More precise deps management for rpm build

bcornec opened the pull request.

bcornec committed:
- RPM build needs python-future

bcornec committed:
Precise python modules versions needed

for RPM build

bcornec committed:
Fix RPM build for 0.3

- Adds entry for management of /var/log/python-redfish by the RPM (not
  used I think it shouldn't be handled that way)
- python-requests >= 2.9.1 is a mandatory installation dep
- Removal for now of the python3 compatible line causing problems with
  setuptools

uggla commented on requirements.txt:
```diff
@@ -8,7 +8,8 @@ Jinja2>=2.7.3
 # Python3 compat
 future>=0.15.2
-configparser>=3.3.0; python_version < '3'
+#configparser>=3.3.0; python_version < '3'
```
configparser is required and should not be commented for python2.
This issus is  "python_version < '3' " that need a really recent version of python setuptools.
I think that the current way to manage that is maybe to make a quick patch with sed in the .spec file.
So remove that "python_version < '3' " for python2 pkg and remove the complete line for python3 pkg.

bcornec replied:
The problem with messing up with this in the .spec is that we need to manage different cases depending of the version of setuptools adopted by the distribution. One way to solve that cleanly would be to use a PB macro in requirements.txt that will be instantiated based on the distro and thus create the right file. Or a macro in the spec doing this. I'll try that.

bcornec replied:
BTW for me the tool is working _without_ having a python-configparser tool !! So for which feature do we need it ?

bcornec replied:
It even doesn't exist in Mageia at all :-(

bcornec replied:
To be mor precise:

$ rpm -qf /usr/lib/python2.7/site-packages/configparser/**init**.py
python-future-0.15.2-1.mga5
$ rpm -qf /usr/lib64/python2.7/ConfigParser.py
python-2.7.9-2.mga5

So do we really need a separate module or is it part of python ?

uggla replied:
I thought about that. Mageia has embeded configparser into future.
But this is not the case for fedora and mainly pypi. (see below)

I'm not opposed to put PBMACRO into requirements.txt, but it means that you will have to generate the pypi stuff. So right now I would not go in this direction.
The sed trick is fine, using it we just manage to have a requirement.txt compatible with all the setuptools version and I think all distro . For 0.3 I will go for this.
More over this problem will not remain anymore as soons as distro will embed a recent setuptools.

```
[uggla@ugglalaptop ~]$ pip show configparser
You are using pip version 7.1.0, however version 8.1.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
---
Metadata-Version: 1.1
Name: configparser
Version: 3.5.0b2
Summary: This library brings the updated configparser from Python 3.5 to Python 2.6-3.5.
Home-page: http://docs.python.org/3/library/configparser.html
Author: Lukasz Langa
Author-email: lukasz@langa.pl
License: MIT
Location: /usr/lib/python2.7/site-packages
Requires: 

```

```
[uggla@ugglalaptop ~]$ rpm -qf /usr/lib/python2.7/site-packages/configparser.py
python-configparser-3.5.0b2-0.2.fc23.noarch
```

bcornec replied:
Today only fedora 23 provides a recent enough version of python-setuptools (18.0.1) to support the syntax python < '3'

bcornec committed:
New version of PR #38

- Put back python funny syntax in requirements.txt
- Create a PB macro (for now in all.pbf) to alter the requirements.txt
  to remove the funny python syntax
